### PR TITLE
Change unit test to use rspec-given

### PR DESCRIPTION
**Pull request by eldriclim**
https://trello.com/c/gaC798My

## Why is this change necessary?

- To get use to rspec-given as it is used extensively in backend
- Improve readability of spec files

## What side effects does this change have?

- None

### CODE REVIEW

File: spec/controllers/assignments_controller_spec.rb [hunk 1]
```diff
@@ -3,60 +3,55 @@
...
-  end
+  }
 
   # Test create action
   describe "post #create" do
-    before do
-      @deliverer = FactoryGirl.create_list(:deliverer, 2)
-      @shift = FactoryGirl.create(:shift)
-    end
+    Given!(:deliverer){ FactoryGirl.create_list(:deliverer, 2) }
```
Comment (leanhdaovn) [inline spec/controllers/assignments_controller_spec.rb:13]: We should favor `Given` as the value is only calculated on the first call, which helps avoid setting up unnecessary data for some context.
`Given!` is used when we need the setup but the variable is not directly called in the subsequent statements.
(This is just FYI. It's not necessary to fix here.)

File: spec/controllers/deliverers_controller_spec.rb [hunk 2]
```diff
@@ -3,95 +3,83 @@
...
+      When {
+        post :create,
           params: { deliverer: FactoryGirl.attributes_for(:deliverer) },
           as: :json
-        }.to change{ Deliverer.count }.by(1)
+        }
 
-        is_expected.to redirect_to home_path
-      end
+      Then { expect(Deliverer.count).to eq 1 }
```
Comment (leanhdaovn) [inline spec/controllers/deliverers_controller_spec.rb:28]: We can also use natural assertions in Then clause. HB Backend follow that style.
https://github.com/jimweirich/rspec-given#natural-assertions
(again, just FYI)

**Review by leanhdaovn — APPROVED** [review]
Good job adopting the style 👍

File: spec/controllers/assignments_controller_spec.rb (hunk 1 above)
Comment (eldriclim) [reply spec/controllers/assignments_controller_spec.rb:13]: Okay I'll take note of that in the future

File: spec/controllers/deliverers_controller_spec.rb (hunk 2 above)
Comment (eldriclim) [reply spec/controllers/deliverers_controller_spec.rb:28]: I'll take note of this too!

**Pull request merged**